plaso/parsers/setupapi.py

```diff
@@ -0,0 +1,205 @@
+      pyparsing.SkipTo(']').setResultsName('exit_status') +
+      pyparsing.SkipTo(pyparsing.lineEnd()) +
+      pyparsing.ZeroOrMore(pyparsing.lineEnd()))
+
+  LINE_STRUCTURES = [
+      ('logline', _SETUPAPI_LINE),
+  ]
+
+  def _GetISO8601String(self, structure):
+    """Retrieves an ISO 8601 date time string from the structure.
+
+    Args:
+      structure (pyparsing.ParseResults): structure of tokens derived from a
+          log entry.
+
+    Returns:
+      str: ISO 8601 date time string.
+
+    Raises:
+      ValueError: if the structure cannot be converted into a date time string.
+    """
+    year = self._GetValueFromStructure(structure, 'year')
+    month = self._GetValueFromStructure(structure, 'month')
+    day_of_month = self._GetValueFromStructure(structure, 'day')
+    hours = self._GetValueFromStructure(structure, 'hours')
+    minutes = self._GetValueFromStructure(structure, 'minutes')
+    seconds = self._GetValueFromStructure(structure, 'seconds')
+    microseconds = self._GetValueFromStructure(structure, 'microseconds')
+
+    try:
+      iso8601 = (
+          '{0:04d}-{1:02d}-{2:02d}T{3:02d}:{4:02d}:{5:02d}.{6:03d}').format(
+              year, month, day_of_month, hours, minutes, seconds, microseconds)
+    except (TypeError, ValueError) as exception:
+      raise ValueError(
+          'unable to format date time string with error: {0!s}.'.format(
+              exception))
+
+    return iso8601
+
+  def _ParseRecordLogline(self, parser_mediator, structure):
+    """Parses a logline record structure and produces events.
+
+    Args:
+      parser_mediator (ParserMediator): mediates interactions between parsers
+          and other components, such as storage and dfvfs.
+      structure (pyparsing.ParseResults): structure of tokens derived from
+          log entry.
+    """
+    date_time = dfdatetime_time_elements.TimeElementsInMilliseconds()
+
+    time_elements_structure = self._GetValueFromStructure(
+        structure, 'start_time')
+    try:
+      datetime_iso8601 = self._GetISO8601String(time_elements_structure)
+      date_time.CopyFromStringISO8601(datetime_iso8601)
+      # Setupapi logs record in local time
+      date_time.is_local_time = True
+    except ValueError:
+      parser_mediator.ProduceExtractionWarning(
+          'invalid date time value: {0!s}'.format(time_elements_structure))
+      return
+
+    # Replace newlines with spaces in structure.message to preserve output.
+    message = self._GetValueFromStructure(structure, 'message')
+    if message:
+      message = message.replace('\n', ' ')
+
+    event_data = SetupapiLogEventData()
+    event_data.entry_type = self._GetValueFromStructure(structure, 'entry_type')
+    # event_data.message = self._GetValueFromStructure(structure, 'message')
+    # event_data.section_end = None
+    event_data.exit_status = self._GetValueFromStructure(
+        structure, 'exit_status')
+
+    event = time_events.DateTimeValuesEvent(
```

Comment:
Since this is local time you want to pass the time zone as in the winfirewall parser
https://github.com/log2timeline/plaso/blob/c9e447abd08f05809ee76e9d1ecd28ef228b60f7/plaso/parsers/winfirewall.py#L186